sklearn/tests/test_isotonic.py

```diff
@@ -166,6 +166,30 @@ def test_isotonic_regression_ties_secondary_():
     assert_array_almost_equal(ir.fit_transform(x, y), y_true, 4)
 
 
+def test_isotonic_regression_with_ties_in_differently_sized_groups():
+    """
+    Non-regression test to handle issue 9432:
+    https://github.com/scikit-learn/scikit-learn/issues/9432
+
+    Compare against output in R:
+    > library("isotone")
```

Comment:
Not an R expert, but I would use something like this to make sure that the `isotone` package is installed:
```R
require('isotone') || (install.packages('isotone') && require('isotone'))
```